torchtune/utils/metric_logging.py

```diff
@@ -160,27 +174,63 @@ def __init__(
                 "Alternatively, use the ``StdoutLogger``, which can be specified by setting metric_logger_type='stdout'."
             ) from e
         self._wandb = wandb
-        self._wandb.init(
-            project=project,
-            entity=entity,
-            group=group,
-            reinit=True,
-            resume="allow",
-            config=kwargs,
-        )
+
+        # logging strategy options are "main", "node", "all"
+        self.log_strategy = log_strategy
+        self.world_size, self.rank = get_world_size_and_rank()
+        self.local_rank = _get_local_rank()
+        self.local_rank = 0 if self.local_rank is None else self.local_rank
+        
+        if (
+                (self.log_strategy == "main" and self.rank == 0) 
+                or (self.log_strategy == "node" and self.local_rank == 0)
+                or self.log_strategy == "all"
+            ): 
+            self._wandb.init(
+                project=project,
+                entity=entity,
+                group=group,
+                reinit=True,
+                resume="allow",
+                **kwargs,
+            
+            )
+
+    def log_config(self, config: DictConfig) -> None:
+        "Logs the config to W&B. Also updates config on overview tab."
+        if self._wandb.run:
+            resolved = OmegaConf.to_container(config, resolve=True)
+            self._wandb.config.update(resolved)
+            
+            output_config_fname = Path(os.path.join(
+                config.checkpointer.checkpoint_dir, 
+                f"torchtune_config_{self._wandb.run.id}.yaml"
+                )
+            )
+            OmegaConf.save(config, output_config_fname)
+            try:
+                
+                log.info(f"Logging {output_config_fname} to W&B under Files")
+                self._wandb.save(output_config_fname, base_path=output_config_fname.parent)
+
+            except Exception as e:
+                log.warning(f"Error saving {output_config_fname} to W&B.\nError: \n{e}")
 
     def log(self, name: str, data: Scalar, step: int) -> None:
-        self._wandb.log({name: data}, step=step)
+        if self._wandb.run:
```

Comment:
Dumb q: what is this doing?

Reply:
wandb won't be initialized if the main/node/rank 0 logic doesn't pass